hibernate-core/src/main/java/org/hibernate/engine/spi/ActionQueue.java

```diff
@@ -105,7 +106,7 @@
 	 */
 	private static final LinkedHashMap<Class<? extends Executable>,ListProvider> EXECUTABLE_LISTS_MAP;
 	static {
-		EXECUTABLE_LISTS_MAP = new LinkedHashMap<Class<? extends Executable>,ListProvider>( 8 );
+		EXECUTABLE_LISTS_MAP = new LinkedHashMap<>( CollectionHelper.determineProperSizing( 8 ) );
```

Comment:
The above usage is a common anti-pattern which almost always ends up with rehashing.

Reply:
Are you sure? when hinted at the initial-capacity, this specific constructor already does a similar calculation.

If you're right that there are improvements to be done here, I'd still prefer to see them in a separate issue (and commit). I hope you don't mind, the reason is the fix seems very important but it also has become very large.

Reply:
Ok I see now the fix is actually very small, sorry I got scared by seeing "27 files changed" ...